Add HostBuilder to 3.0 migration docs (#10718)
aspnet/Docs: aspnetcore/migration/22-to-30.md

               options.SerializerSettings.ContractResolver = 
                  new CamelCasePropertyNamesContractResolver());
    ```

> ## Generic host replaces WebHostBuilder

Reviewer: Either ...

Generic Host replaces Web Host

or

HostBuilder replaces WebHostBuilder

or

Generic Host (HostBuilder) replaces Web Host (WebHostBuilder)

> The ASP.NET Core 3.0 templates use [Generic Host](fundamentals/host/generic-host). Previous versions used [WebHostBuilder](xref:fundamentals/host/web-host). The following code shows the ASP.NET Core 3.0 template generated `Program` class:

Reviewer: ```suggestion
The ASP.NET Core 3.0 templates use the [Generic Host](xref:fundamentals/host/generic-host). Previous versions used the [Web Host](xref:fundamentals/host/web-host). The following code shows the ASP.NET Core 3.0 template-generated `Program` class:
```

> [!code-csharp[Main](22-to-30/samples/Program.cs?name=snippet)]

Reviewer: ```suggestion
[!code-csharp[](22-to-30/samples/Program.cs?name=snippet)]
```

> The following code shows the ASP.NET Core 2.2 template generated `Program` class:

Reviewer: ```suggestion
The following code shows the ASP.NET Core 2.2 template-generated `Program` class:
```

> [!code-csharp[Main](22-to-30/samples/Program2.2.cs?name=snippet)]

Reviewer: ```suggestion
[!code-csharp[](22-to-30/samples/Program2.2.cs?name=snippet)]
```

> <xref:Microsoft.AspNetCore.Hosting.IWebHostBuilder> is used with the <xref:Microsoft.AspNetCore.Hosting.WebHostBuilder> (in the 2.2 templates) and with `webBuilder` (in the 3.0 templates). `WebHostBuilder` will be deprecated in a future release, and eventually removed.  `HostBuilder` will replace `WebHostBuilder`.

Reviewer: Used with? ... `WebHostBuilder` implements `IWebHostBuilder` to build an `IWebHost`.

I can't review all of the first sentence ... I need to dig into it more because the sand has perhaps shifted a bit since 2.2 (it was/is `HostBuilder` implements `IHostBuilder` to build an `IHost`). I've been 🏃 on other issues. If you want to wait until Thursday, I can take a closer look and then review these remarks.

Last two sentences ok, but there's a comma splice (before "and") and an extra space after "removed."

> ## Moving from WebHostBuilder to HostBuilder

Reviewer: ```suggestion
## Move from WebHostBuilder to HostBuilder
```

> The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). `HostBuilder` can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration> and <xref:Microsoft.Extensions.Hosting.IHostingEnvironment>. The `HostBuilder` DI limitations:

Reviewer 2: ```suggestion
The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). When using `HostBuilder` you can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration>, <xref:Microsoft.Extensions.Hosting.IHostingEnvironment>, and <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment> into Startup's constructor. The `HostBuilder` DI constraints:
```

Reviewer: Comma  ....

> When using \`HostBuilder\`, you can ...

maybe "**the** Startup's constructor" ?

> The most significant change from `WebHostBuilder` to `HostBuilder` is in [Dependency injection](xref:fundamentals/dependency-injection) (DI). `HostBuilder` can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration> and <xref:Microsoft.Extensions.Hosting.IHostingEnvironment>. The DI limitations:

Reviewer: ```suggestion
The most significant change from `WebHostBuilder` to `HostBuilder` is in [dependency injection (DI)](xref:fundamentals/dependency-injection). `HostBuilder` can only inject <xref:Microsoft.Extensions.Configuration.IConfiguration> and <xref:Microsoft.Extensions.Hosting.IHostingEnvironment>. The DI limitations:
```

* Enable the DI container to be built only one time.

> * Avoid several other DI issues.

Reviewer 2: ```suggestion
* Avoids the resulting object lifetime issues like resolving multiple instances of singletons.
```

Reviewer: Prefer "... , such as ..." for examples.

> * Enable the DI container to built once.

Reviewer: ```suggestion
* Enable the DI container to build once.
```

?? I don't understand the remark.

Author: changed it to
enable the DI container to be built only one time.

> * Avoid several other DI issues.

Reviewer: Nonspecific ... I guess engineering didn't provide details. Wish for examples or something.
